hudi-hadoop-mr/src/main/java/org/apache/hudi/hadoop/HoodieCopyOnWriteTableInputFormat.java

```diff
@@ -189,11 +190,16 @@ protected List<FileStatus> listStatusForIncrementalMode(JobConf job,
     return HoodieInputFormatUtils.filterIncrementalFileStatus(jobContext, tableMetaClient, timeline.get(), fileStatuses, commitsToCheck.get());
   }
 
-  protected FileStatus createFileStatusUnchecked(FileSlice fileSlice, HiveHoodieTableFileIndex fileIndex, Option<HoodieVirtualKeyInfo> virtualKeyInfoOpt) {
+  protected Option<? extends FileStatus> createFileStatusUnchecked(FileSlice fileSlice, HiveHoodieTableFileIndex fileIndex, Option<HoodieVirtualKeyInfo> virtualKeyInfoOpt) {
     Option<HoodieBaseFile> baseFileOpt = fileSlice.getBaseFile();
+    Option<HoodieLogFile> latestLogFileOpt = fileSlice.getLatestLogFile();
 
     if (baseFileOpt.isPresent()) {
       return getFileStatusUnchecked(baseFileOpt.get());
+    } else if (latestLogFileOpt.isPresent()) {
+      // It happens when reading optimized query to mor.
+      LOG.info("File slice(" + fileSlice.getFileId() + ") has no base-file but log-file. Skip the slice.");
+      return Option.empty();
```

Comment:
Add these lines to avoid IllegalStateException, when reading optimized query to mor with file slice without base file

Reply:
with hbase index, its feasible to have only log files w/o a base file on some cases. its also the case for kafka connect. So, its valid case even for snapshot query if I am not wrong. so, looks like this fix was needed anyways even if not for read optimized query.
@alexeykudinkin @yihua

Reply:
That's true, let's make this fix into release 0.12.0